app/resolver.go

```diff
@@ -15,9 +15,11 @@ var (
 	lookupIP = net.LookupIP
 )
 
-// Resolver periodically tries to resolve the IP addresses for a given
-// set of hostnames.
-type Resolver struct {
+type resolver interface {
```

Comment:
It maintains consistency with other, similar components e.g. the reporter, and it bounds the surface area of the component at its callsite. More generally, it's best if constructors return interface types rather than pointers to implementations, as it forces you to think about APIs.